reference/reference_test.go

```diff
@@ -273,3 +273,57 @@ func TestInvalidReferenceComponents(t *testing.T) {
 		t.Fatal("Expected WithName to detect invalid digest")
 	}
 }
+
+func TestMatch(t *testing.T) {
+	matchCases := []struct {
+		reference string
+		pattern   string
+		expected  bool
+	}{
+		{
+			reference: "foo",
+			pattern:   "foo/**/ba[rz]",
```

Comment:
The globbing in golang's stdlib doesn't support `**` for multi-component match. This case is still valid but this pattern will only match `foo/a/ba[rz]`, not `foo/a/b/ba[rz]`. I don't think we need to do more for this PR.